exporter/datadogexporter/internal/logs/translator.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/logs"
+
+import (
+	"encoding/binary"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/attributes"
+	"github.com/DataDog/datadog-agent/pkg/otlp/model/source"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadog"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+)
+
+const (
+	namespace          = "otel"
+	otelTraceID        = namespace + ".trace_id"
+	otelSpanID         = namespace + ".span_id"
+	otelSeverityNumber = namespace + ".severity_number"
+	otelSeverityText   = namespace + ".severity_text"
+	otelTimestamp      = namespace + ".timestamp"
+
+	ddStatus    = "status"
+	ddTraceID   = "dd.trace_id"
+	ddSpanID    = "dd.span_id"
+	ddTimestamp = "@timestamp"
+)
+
+const (
+	logLevelTrace = "trace"
+	logLevelDebug = "debug"
+	logLevelInfo  = "info"
+	logLevelWarn  = "warn"
+	logLevelError = "error"
+	logLevelFatal = "fatal"
+)
+
+// Transform is responsible to convert LogRecord to datadog format
+func Transform(lr plog.LogRecord, res pcommon.Resource, sendLogRecordBody bool) datadogV2.HTTPLogItem {
+	hostName, serviceName := extractHostNameAndServiceName(res.Attributes(), lr.Attributes())
+
+	l := datadogV2.HTTPLogItem{
+		AdditionalProperties: make(map[string]string),
+	}
+	if hostName != "" {
+		l.Hostname = datadog.PtrString(hostName)
+	}
+
+	if serviceName != "" {
+		l.Service = datadog.PtrString(serviceName)
+	}
+
+	// we need to set log attributes as AdditionalProperties
+	lr.Attributes().Range(func(k string, v pcommon.Value) bool {
+		l.AdditionalProperties[k] = v.AsString()
+		return true
+	})
+	if !lr.TraceID().IsEmpty() {
+		l.AdditionalProperties[ddTraceID] = strconv.FormatUint(traceIDToUint64(lr.TraceID()), 10)
+		l.AdditionalProperties[otelTraceID] = lr.TraceID().HexString()
```

Comment:
How is this going to work once we fully migrate to 128-bit? Is `otel.trace_id` going to be our 128-bit trace ID key forever? Why not just use `trace_id` for the 128-bit value? It's the W3C standard and what we are migrating too anyway... Arguably what you have now makes it consistent with APM. Just a thought.